backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthenticationFilter.java

```diff
@@ -53,7 +54,9 @@ public Authentication attemptAuthentication(
       if (loginAttemptService.hasTooManyAttempts(userDetails.getUsername()))
         throw new TooManyAttemptsException();
 
-      if (!authService.isActive(userDetails.getUsername())) throw new EmailNotVerifiedException();
+      if (!authService.isVerified(userDetails.getUsername())) throw new EmailNotVerifiedException();
+
+      if (!authService.isActive(userDetails.getUsername())) throw new UserNotActiveException();
```

Comment:
is this revealing too much info? should we just return bad credentials or something?

Reply:
My logic was, the user should know when he needs to confirm his email to log in, and in the other case when he is banned, he should know there's nothing he can do.

Reply:
makes sense, although most websites would probably send an email notification or something if you try to use a banned account

but this is probably fine for our use case